zebra-state/src/service/check/anchors.rs

```diff
@@ -159,10 +97,231 @@ pub(crate) fn anchors_refer_to_earlier_treestates(
             {
                 return Err(ValidateContextError::UnknownOrchardAnchor {
                     anchor: orchard_shielded_data.shared_anchor,
+                    height: prepared.height,
+                    tx_index_in_block,
+                    transaction_hash: prepared.transaction_hashes[tx_index_in_block],
                 });
             }
 
-            tracing::debug!(?orchard_shielded_data.shared_anchor, "validated orchard anchor");
+            tracing::debug!(
+                ?orchard_shielded_data.shared_anchor,
+                ?tx_index_in_block,
+                height = ?prepared.height,
+                "validated orchard anchor",
+            );
+        }
+    }
+
+    Ok(())
+}
+
+/// This function fetches and returns the Sprout final treestates from the state,
+/// so [`sprout_anchors_refer_to_interstitial_treestates()`] can check Sprout
+/// final and interstitial treestates without accessing the disk.
+///
+/// Sprout anchors may also refer to the interstitial output treestate of any prior
+/// `JoinSplit` _within the same transaction_.
```

Comment:
I was a bit confused about these two functions but I get it now, here's a suggestion to hopefully make it clearer, feel free to modify

```suggestion
/// Sprout anchors may also refer to the interstitial output treestate of any prior
/// `JoinSplit` _within the same transaction_; these are created on the fly
/// in [`sprout_anchors_refer_to_treestates`].
```

Reply:
Applied in eebf3f581b